posts/copilot-workspace/index.qmd

```diff
@@ -0,0 +1,93 @@
+
+### Outcome
+
+Here, Copilot provided no analysis of the code base. It regurgitated the changes described in the [package review pull request](https://epiverse-trace.github.io/blueprints/code-review.html#full-package-review), likely from a mix of the pull request description and the `NEWS.md` file, and did not analyse any of the code. Therefore, it provided no additional use than if the reviewer had just read the `NEWS`.
+
+## Conclusion
+
+We think there’s still a long way to go before AI-powered tools produce code projects unsupervised that are equivalent to that from a research software engineer. One issue is the generative AI has training database that leans towards common problems, which can makes its solutions to coding problems with specific constraints suboptimal, particularly in fields where there is a relatively small public code base to train models with (e.g. R functions for epidemic analysis are much rarer than generic JavaScript functions for web design). A second is that AI-models are picking up common patterns and synthesising them, and didn’t seem to be able to understand how different parts of a codebase link together, which a developer would be able to utilise to reduce duplication and keep a consistent design pattern. Put together, we think the applicability of current generative AI models to aid research software engineering in domain-specific R package development, where bespoke solutions are common, is likely to be low. This could improve for use-cases which seek to implement more common tasks, such as building dashboards. The test cases we experimented with demonstrate that this tool will not replace, and at times not even enhance development when the task requires a more integrated development than generating a standalone script. These tools are evolving rapidly and we are sure improvements will continue. In the short term tools like GitHub copilot workspace need to be used responsibly with an awareness of their limitations and not a blind trust that the code it writes is always correct. We would caution against relying on generative AI tools alone to develop software with potentially large-scale impacts - scrutiny by trained developers is still key. Research software engineers will play a key part in this as they will ensure that code and software written with and by AI tools meet the right standards, are trustworthy, and are not potentially harmful. Where we see generative AI tools being useful in research software development is in [tasks that require sifting through or summarising vast amounts of information](https://billyquilty.substack.com/p/llmlinelist-enabling-rapid-outbreak); for example, in querying software documentation, or to help get started with a new programming framework or language.
```

Comment:
```suggestion
The evidence above suggests there’s still a long way to go before AI-powered tools produce code projects unsupervised that are equivalent to that from a research software engineer. One issue is the generative AI has training database that leans towards common problems, which can makes its solutions to coding problems with specific constraints suboptimal, particularly in fields where there is a relatively small public code base to train models with (e.g. R functions for epidemic analysis are much rarer than generic JavaScript functions for web design). A second is that AI-models are picking up common patterns and synthesising them, and didn’t seem to be able to understand how different parts of a codebase link together, which a developer would be able to use to reduce duplication and keep a consistent design pattern. Put together, we think the applicability of current generative AI models to aid research software engineering in domain-specific R package development, where problems can be complex bespoke solutions are common, is likely to be low. This could improve for use-cases which seek to implement more common tasks, such as building dashboard templates. 

The test cases we experimented with suggest that this tool will not replace, and at times not even enhance development when the task requires a more integrated development than generating a standalone script. These tools are evolving rapidly and we are sure improvements will continue. In the short term tools like GitHub copilot workspace need to be used responsibly with an awareness of their limitations and not a blind trust that the code it writes is always correct. We would caution against relying on generative AI tools alone to develop software with potentially large-scale impacts - scrutiny by trained developers is still key. Research software engineers will play a key part in this as they will ensure that code and software written with and by AI tools meet required standards, are trustworthy, and are not potentially harmful. Where we see generative AI tools being more immediately useful in research software development is in [tasks that require sifting through or summarising vast amounts of information](https://billyquilty.substack.com/p/llmlinelist-enabling-rapid-outbreak); for example, in querying software documentation, or to help get started with a new programming framework or language.
```
As 'demonstrate' is quite strong and 'use' generally preferable to 'utilise'. Also a few tweaks for readability.